llvmlite/tests/test_binding.py

```diff
@@ -982,5 +982,47 @@ def test_libm(self):
         llvm.load_library_permanently(libm)
 
 
+class TestGlobalVariables(BaseTest):
+    def check_global_variable_linkage(self, linkage):
+        mod = ir.Module()
```

Comment:
The main reason is that it is not clear in the llvm langref spec which linkage requires an initializer.  I am relying on parse_assembly() to validate the IR.